### PR TITLE
Handles custom build types when adding Flutter jar dependencies to a Flutter Gradle Plugin

### PR DESCRIPTION
**Pull request by josh-burton**
Flutter Gradle Plugin

## Description

A project with a custom build type would fail to build if it used plugins.

The custom build types propagate down to the plugin projects, but the plugin projects were only adding the Flutter jar to a fixed set of build types.

The Flutter jar is now added to all build types.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/gradle/flutter.gradle [hunk 1]
```diff
@@ -196,42 +196,57 @@ class FlutterPlugin implements Plugin<Project> {
...
         }
     }
 
-    private void addFlutterJarCompileOnlyDependency(Project project) {
+    /**
+     * Adds suitable flutter.jar api dependencies to the specified buildType.
+     *
+     * Note: The BuildType DSL type is not public, and is therefore omitted from the signature.
+     */
+    private void addFlutterJarCompileOnlyDependency(Project project, buildType) {
```
Comment (athornz) [inline packages/flutter_tools/gradle/flutter.gradle]: I'm unsure if this method was originally designed to handle non-Android plugin projects. Is that a valid usecase?

These changes are currently dependent on the plugin being an Android project.

**Comment by athornz** [comment]
@dnfield Wondering if you are the right person to review this PR?

**Comment by jonahwilliams** [comment]
We can't accept any PRs to the tool or gradle scripts without a way to test it and prevent regressions. In this case, it will likely require a project configured accordingly under `dev/integration_tests` and a corresponding test case is `dev/devicelab`

**Comment by athornz** [comment]
Thanks @jonahwilliams I'll work on adding some tests.

**Comment by jonahwilliams** [comment]
Here is an example of one of these devicelab tests: https://github.com/flutter/flutter/blob/master/dev/devicelab/bin/tasks/build_mode_test.dart

This runs a particular app from the integration_tests directory

**Pull request closed without merging**